frame/staking/src/pallet/mod.rs

```diff
@@ -850,18 +849,16 @@ pub mod pallet {
 		#[pallet::weight(T::WeightInfo::bond())]
 		pub fn bond(
 			origin: OriginFor<T>,
-			controller: AccountIdLookupOf<T>,
 			#[pallet::compact] value: BalanceOf<T>,
 			payee: RewardDestination<T::AccountId>,
 		) -> DispatchResult {
 			let stash = ensure_signed(origin)?;
+			let controller = stash.clone();
```

Comment:
I'd probably rename this to `controller_deprec` or something, but since this will eventually be removed should be fine for now.

Reply:
Or maybe why not:

```rust 
if <Ledger<T>>::contains_key(&stash_or_controller) {
  return Err(Error::<T>::AlreadyPaired.into())
}

// ...

Self::update_ledger(&stash, &item);
```

then we could remove the `controller` altogether.